sdk/core/azure-core/inc/azure/core/datetime.hpp

```diff
@@ -76,14 +93,42 @@ namespace Azure { namespace Core {
         std::string const& timeString,
         DateFormat format = DateFormat::Rfc1123);
 
+  private:
     /**
      * @brief Get a string representation of the @DateTime.
      *
      * @param format The representation format to use.
+     * @param fractionFormat The format for the fraction part of the Datetime. Only supported by
+     * ISO8601
      *
      * @throw DateTimeException If year exceeds 9999, or if \p format is not recognized.
      */
-    std::string ToString(DateFormat format = DateFormat::Rfc1123) const;
+    std::string ToString(DateFormat format, TimeFractionFormat fractionFormat) const;
+
+  public:
+    /**
+     * @brief Get a string representation of the @DateTime.
+     *
+     * @param format The representation format to use.
+     *
+     * @throw DateTimeException If year exceeds 9999, or if \p format is not recognized.
+     */
+    std::string ToString(DateFormat format = DateFormat::Rfc1123) const
+    {
+      return ToString(format, TimeFractionFormat::DropTrailingZeros);
+    };
+
+    /**
+     * @brief Get a string representation of the @DateTime formated with ISO8601.
+     *
+     * @param fractionFormat The format that is applied to the fraction part from the ISO8601 date.
+     *
+     * @throw DateTimeException If year exceeds 9999, or if \p fractionFormat is not recognized.
+     */
+    std::string ToIso8601String(TimeFractionFormat fractionFormat) const
+    {
+      return ToString(DateFormat::Iso8601, fractionFormat);
```

Comment:
Validate input `TimeFractionFormat` enum is within the valid range and add test.

Reply:
valid range for enum?
All the enumarated values from the enum are valid. I don't understand what you say

Reply:
```C++
#include <iostream>

enum class TimeFractionFormat
{
    DropTrailingZeros,
    AllDigits,
    Truncate
};

int valid(TimeFractionFormat num) {
    if (num == TimeFractionFormat::DropTrailingZeros)
    {
        return 1;
    }
    if (num == TimeFractionFormat::AllDigits)
    {
        return 2;
    }
    if (num == TimeFractionFormat::Truncate)
    {
        return 3;
    }
    return -1;
}

int main()
{
    // emits -1 because the enum valule is invalid
    // We should be checking for enum range and throw.
    std::cout << valid((TimeFractionFormat)5);
}

```